Blog post: CSL4LibreOffice (#97)
JabRef/blog.jabref.org: _posts/2024-08-23-GSoC-CSL.md

## Background

As a student, I first came across pre-defined citation formats during my freshman year of college, when our professor instructed that the references in our project report must follow the "APA style".  

I remember manually generating and copying citations in that style from some citation generation website, and oh was it tedious. Whether you are a student like me or a seasoned researcher in academia, you must have grappled with various citation styles throughout your academic journey.

[Citation Style Language](https://citationstyles.org/) (CSL) is a popular open-source specification language that standardizes the formatting of citations and bibliographies. It supports thousands of standard citation styles used in academia, including *American Psychological Association* (APA), *Modern Language Association* (MLA), *Chicago Manual of Style* (CMS), *Institute of Electrical and Electronics Engineers* (IEEE), *Springer - Lecture Notes in Computer Science* and *Vancouver* to name a few. If you've ever cited a source in an academic paper, chances are you've used a CSL-based style.

> For several years, JabRef users have been requesting support for CSL in its LibreOffice integration. Tackling this problem statement this summer as a part of my project, I am excited to announce that this highly anticipated feature is now available!

Reviewer: its --> JabRef's

## What's New?

With this integration, JabRef users can now:

> 1. **Choose a CSL Style**: Select from a vast array of CSL styles to match the requirements of your academic work.

Reviewer: array? Maybe "collection"?

> 2. **Cite with Flexibility**: Insert citations as well as in-text citations (similar to LaTeX's `\cite` and `\citet` commands) directly into your LibreOffice document.

Reviewer: "Insert usual citiations"...

It reads strange that no attribute for "citations" first, but we have no word for "non-intext.citations", do we?

Replace "LaTeX's" by "natbib's" and link https://ctan.org/pkg/natbib.

Author: "Usual citations" sounds odd.
How about this ^ (check changes).

Author: If not this, the other alternative - "insert citations directly into..." (remove in-text, as citations is generic and can encompass all types).

Reviewer: Then also remove link to natbib, because there is no distinction any more

Author: Sorry, what I meant was, I have already changed it to:
"**Cite with Flexibility**: Insert citations (similar to natbib's `\cite`) as well as in-text citations (similar to natbib's and `\citet`) directly into your LibreOffice document."

If we don't like this, then we can go with:
"**Cite with Flexibility**: Insert citations directly into your LibreOffice document."
(with natbib removed).

Author: I realized I forgot to add the link. Will wait for your approval (if we choose to go with the first one).

Author: I think Thilo isn't tagged properly so @ThiloteE

Author: @koppor Side comment: I do think case 2 looks cleaner (and more digestible for a general audience, even for students like me who didn't use LaTeX during the college course, the story of which I mentioned in the blog's introductory paragraph).

Reviewer 2: I suggest trying to avoid filler words and non-words, such as "flexible", especially for documentation. I think the sentence is good enough for the Blog, though. While I have used LaTeX and bibaltex, I have never used the Natbib package before. I think JabRef and CSL define the way they cite themselves independently from Natbib, and, yes, while it can help users that are used to that package help understand the new features it might be easier to understand, if the feature is explained, with verbs instead of nouns.

Author: Which sentence, Thilo?
Your suggestion would be to remove mentioning natbib (or LaTeX) altogether?
Are you referring to case 1, case 1 with natbib removed, or case 2?

Author: I have updated it to:
"**Cite with ease**: Insert citations directly into your LibreOffice document."

Reviewer 2: I am not sure what case 1 or case 2 are.

Ideally, I would prefer splitting up Cite and Cite-in-Text into separate sections, but it's fine, if that is explained more in detail in the documentation. 
The way it is now with natbib is ok. We have many LaTeX users, so Natbib might ring a bell.

Author: Can you check if the current sentence is okay? Or if I should revert it and bring back the distinctions and natbib refrences?

Reviewer 2: To be honest, I don't have a strong opinion here. Your choice :-)

Author: Alright.
@koppor Then I suggest the current going with the current:
"Cite with ease: Insert citations directly into your LibreOffice document."

3. **Auto-generate Bibliographies**: Watch as your bibliography automatically populates based on your citations, with just a single click.
4. **Update Bibliographies**: Easily refresh your bibliography as you add new citations to your document.

## Getting Started

To start using this new feature:

> 1. Download the [development version of JabRef](https://builds.jabref.org/main/)

Reviewer 2: 1. Download the [development version of JabRef](https://builds.jabref.org/main/).

> 1. Update to the latest version and start JabRef

Reviewer: ```suggestion
1. Download the [development version of JabRef](https://builds.jabref.org/main/).
```

> 2. Connect to a running LibreOffice document instance from the OpenOffice Panel

Reviewer: ```suggestion
2. Connect to a running LibreOffice document instance from the [Libre/OpenOffice Panel](https://docs.jabref.org/cite/openofficeintegration)
```

> 2. Connect to a running LibreOffice document instance from the [Libre/OpenOffice Panel](https://docs.jabref.org/cite/openofficeintegration)

Reviewer 2: 2. Connect to a running LibreOffice document instance by either clicking the "Connect" or "Manual Connect" button in the [Libre/OpenOffice Panel](https://docs.jabref.org/cite/openofficeintegration) in the side pane.

3. Click on "Select style" from the panel
4. Choose your preferred CSL style from the list and click OK
5. Select an entry (or a group of entries) and click "Cite" to insert citations into the document
6. Click on the "Make/Sync bibliography" on the panel to auto-generate a bibliography section in your document based on your citations and the selected CSL style

### Demo

#### Selecting a style

![Selecting a style](../img/selecting-a-style.gif)

#### Inserting citations

![Inserting citations](../img/cite.gif)

#### Generating bibliography

![Generating bibliography](../img/bibliography.gif)

#### Updating bibliography

![Updating bibliography](../img/refresh.gif)

## Summary

> We hope this new feature enhances your research and writing process. As always, we welcome your feedback and suggestions for further improvements.

Reviewer: Add link to forum for feedback - otherwise, users might not know where to put feedback.

```suggestion
I hope this new feature enhances your research and writing process. As always, [we welcome your feedback and suggestions for further improvements](https://discourse.jabref.org/c/feedback/3).
```

Reviewer: Also move the paragraph as last.

Reason: Structure of a summary is:

1. Summary
2. Future work

If you are interested in the technical details of how this project was undertaken, do check out the [wiki](https://github.com/JabRef/jabref/wiki/GSoC-2024-%E2%80%90-Improved-CSL-Support-%28and-more-LibreOffice%E2%80%90JabRef-integration-enhancements%29).

> I hope this new feature enhances your research and writing process. As always, [we welcome your feedback and suggestions for further improvements](https://discourse.jabref.org/c/feedback/3).

Reviewer: Use passive voice in second part. "We" sounds strange, b/c all other parts are "I". Since you also welcome feedback *too* I suggest passive voice
